lib/ansible/modules/packaging/os/satellite_subscription.py

```diff
@@ -0,0 +1,441 @@
+    Your hosts or hypervisors should be in the Satellite already. The module will run subscription
+    related tasks (assign, remove, autoattach or replace with hypervisor subscriptions) for the
+    hosts through the Satellite API.
+version_added: "2.8"
+requirements:
+  - requests
+author:
+  - Luc Stroobant (@stroobl)
+options:
+  satellite_hostname:
+    description:
+      - Red Hat Satelite server (6.x) hostname
+    required: true
+  username:
+    description:
+      - Satellite server username
+    required: true
+  password:
+    description:
+      - Satellite server password
+    required: true
+  organization:
+    description:
+      - Satellite organization ID to work on
+    required: true
+  state:
+    description:
+      - "present: assign a subscription of subscription_type to host"
+      - "absent: remove subscription of subscription_type from host"
+      - "autoattach: run subscription autoattach for host (autoattach will be enabled if necessary)"
+      - "disable_autoattach: disable subscription autoattach for host"
+      - "vdcguests: remove all Red Hat subscriptions from a hypervisor's guests and run autoattach
+         to replace them with virtual guest subscriptions. Run this against a virt-who hypervisor to
+         remove all other subscriptions assigned to the guests. Useful if you configured a VDC
+         subscription on a hypervisor that didn't have it before."
+      - "optimize: use to replace subscriptions on physical servers with a higher multiplier with subscriptions with a lower multiplier."
+    choices: [present, absent, autoattach, disable_autoattach, vdcguests, optimize]
+    default: autoattach
+  content_host:
+    description:
+      - List of content hosts to work with
+      - Can also be a hostname search pattern accepted by the satellite.
+  subscription:
+    description:
+      - List of subscriptions to assign or remove
+      - Required when action is assign or remove
+      - "This field is used for the search for subscriptions, it can contain (part of) the subscription
+         name, eg 'Red Hat Enterprise Linux Server Entry Level, Self-support' or another search query
+         as used in the satellite, eg: 'product_id=RH00005S'"
+  unique:
+    description:
+     - Set to true to remove all other subscriptions attached to the host
+    type: bool
+    default: False
+  verify_ssl:
+    description:
+      - Check ssl certificates?
+    type: bool
+    default: True
+"""
+
+EXAMPLES = """
+- name: Assign subscriptions, remove all existing subscriptions (run against localhost)
+  satellite_subscription:
+    satellite_hostname: "{{ satellite_fqdn }}"
+    username: "{{ satellite_admin_user }}"
+    password: "{{ satellite_admin_pass }}"
+    state: present
+    content_host: "{{ groups.rhel_servers }}"
+    subscription:
+      - product_id=RH000XX
+      - name=EPEL
+      - name=Extra
+    unique: true
```

Comment:
Interesting. Defaults for molecule/ansible-lint  prefer true/false.

Reply:
Yup, there is an action item to change that. Strict YAML 1.2 only accepts true/false, but normal YAML accepts too many options as booleans (y/n, true/false, yes/no, on/off). Obviously we don't want to have all these in our examples, as this leads to confusion.

We decided that we would allow both yes/no and true/false, but for documentation and examples would stick to yes/no. The main reason for this is that it is easier for non-native English speakers. (Since YAML interpretes the boolean values, we don't know exactly what the DOCUMENTATION specified, so we can't just reproduce what was stated and have to decide on one.)